frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx

```diff
@@ -121,6 +116,12 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
     showSideBar,
     disabled,
   });
+  const { buildingWithParcelDetail } = useSideBarBuildingWithParcelLoader({
```

Comment:
This is the new hook to handle new buildings that are associated to land. Having this in a separate hook helps prevent some of the state contamination that was happening when this logic was part of the building loader.